libsel4/arch_include/riscv/sel4/arch/types.h

```diff
@@ -60,6 +61,22 @@ typedef struct seL4_UserContext_ {
     seL4_Word tp;
 } seL4_UserContext;
 
+typedef struct seL4_RISCV_SBIContext_ {
+    seL4_Word a0;
+    seL4_Word a1;
+    seL4_Word a2;
+    seL4_Word a3;
+    seL4_Word a4;
+    seL4_Word a5;
+    seL4_Word a6;
+    seL4_Word a7;
+} seL4_RISCV_SBIContext;
+
+typedef struct seL4_RISCV_SBIRet_ {
+    seL4_Word error;
+    seL4_Word value;
+} seL4_RISCV_SBIRet;
```

Comment:
Just a suggestion, but wouldn't it make more sense to use the same construction as `seL4_ARM_Page_GetAddress` and combine seL4's errors with SBI errors?

SBI errors are negative, seL4's errors are positive numbers, both use 0 for no error. Or alternatively, use a specific seL4 error code for SBI errors and return the SBI error in `value`.